compiler/ballerina-backend-jvm-old/src/main/ballerina/compiler_backend_jvm/jvm_method_gen.bal

```diff
@@ -638,11 +638,13 @@ function generateBasicBlocks(jvm:MethodVisitor mv, bir:BasicBlock?[] basicBlocks
             } else if (inst is bir:NewMap) {
                 instGen.generateMapNewIns(inst);
             } else if (inst is bir:NewTypeDesc) {
-                instGen.generateNewTypedescIns(inst);
+                 if (inst.kind == bir:INS_KIND_NEW_TYPEDESC) {
```

Comment:
This is due to `NewStream` being structurally equivalent to `NewTypeDesc` I assume?

Wondering if instead, we should update `NewStream` to have a field of a different name (maybe rename `typeValue` as `constraint`) so that it isn't structurally equivalent to `NewTypeDesc`?

Both stream and typedesc going in the same path doesn't seem right?

Reply:
This is not the constraint.. it is the streamType. may be streamType or sType?

Reply:
+1